config/templates/pages/example.xml

```diff
@@ -126,6 +126,21 @@
             </types>
         </block>
 
+        <property name="text_line" type="text_line">
+            <meta>
+                <title lang="en">Text Line</title>
+                <title lang="de">Text</title>
+                <info_text lang="en">There are no commas allowed in this field</info_text>
+                <info_text lang="de">In diesem Feld sind keine Beistriche erlaubt</info_text>
+            </meta>
+
+            <params>
+                <param name="pattern" value="^[^,]*$"/>
+                <param name="min_characters" value="2"/>
+                <param name="max_characters" value="3"/>
```

Comment:
There is already a soft limit on length in the `page_seo` form, why can I still save the page with too many character there, but it is not possible here? Seems a bit confusing to me :thinking:

Reply:
because it's not a `text_line` 🙈

Reply:
Looks like a text line to me :thinking: 
https://github.com/sulu/sulu/blob/cf138452aeb5cb1b252f8921806088e2a8997279/src/Sulu/Bundle/PageBundle/Resources/config/forms/page_seo.xml#L14

Reply:
oops, my fault ... i was thinking of the seo description 😄

This is the schema, which is being generated
```json
{
  "type": "object",
  "properties": {
    "ext/seo/title": {
      "const": "foo"
    }
  }
}
```
and this the corresponding data
```json
{
  "ext": {
    "seo": {
      "title": "bar"
    }
  }
}
```
My guess is that ajv searches for a property called `ext/seo/title`, but that doesn't exist. there's just an `ext` property, which is the object we see above. And since we allow additional properties, the validation passes.

Reply:
Does this override the soft limit? I think we should not override the exist softlimit behaviour as this is by design a soft limit because its just a recommendation. I think it would better to use `minLength` and `maxLength` for hard limit text_line and keep the soft limit as it is, maybe deprecate it and rename it but it should still be possible to have soft limit for the seo things.

Reply:
But then I would rename the `max_characters` to something like `max_character_recommendation`, to make sure, that this is not a hard constraint. And also we should not use the `CharacterCounter` component for the `maxLength` then, to clearly differentiate between those two concepts.

Reply:
But yes, sounds like a good idea to me

Reply:
Yeah, using the `CharacterCounter` felt also weird to me... I think I would let the current param work as it did (no hard constraint) and deprecate it right now. In order to be consistent with the other params I would name the new ones `min` and `max`, and these should be the hard limit, and the form should fail if they happen. The deprecated ones could be replaced with something like `soft_min` and `soft_max`. I think `soft` has enough meaning in order to convey what it means and it is a lot less to type than `max_character_recommendation` (remember that in this place there also is no auto completion).

But I am also open to other suggestions.

Reply:
I don't think I would name these parameters `min` and `max`, because that doesn't describe the resulting constraints very well. For selections it's pretty clear what `min` and `max` is meaning, the same would be for a number field. But I think for text fields i would call it `minLength` and `maxLength`, because there could also be `minSegments` and `maxSegments` ...

Reply:
I would also go with `minLength` and `maxLength`. Because `min` and `max` for the `input` is something different https://www.w3schools.com/tags/tag_input.asp which would be used for `number` input.

Reply:
Fine for me. What about the soft limit? `minSoftLength` and `maxSoftLength`?

Reply:
is okay for me ... but currently we only have `max_characters` but no `min_characters`, so should we really implement a `minSoftLength`?

Reply:
guess i would use `softMinLength`, but `minSoftLength` is okay for me too 🙂

Reply:
>  minSoftLength and maxSoftLength?

Is fine for me.

Reply:
> is okay for me ... but currently we only have `max_characters` but no `min_characters`, so should we really implement a `minSoftLength`?

From my POV we can also leave it out... Not sure how that should be handled if you have set a soft min and max value :thinking: